revamp basic concepts docs (#77748)
ansible/ansible: docs/docsite/rst/shared_snippets/basic_concepts.txt

Control node
============

> The machine from which you run the Ansible CLI tools (with Ansible installed), the ``ansible`` or ``ansible-playbook`` binaries. You can use any computer that meets the software requirements as a control node - laptops, shared desktops, and servers can all run Ansible. You can have multiple control nodes, but Ansible itself does not coordinate across them, see AAP for a .

Reviewer: ```suggestion
You install Ansible on the machine you run the Ansible CLI tools, the ``ansible`` or ``ansible-playbook`` binaries. You can use any computer that meets the software requirements as a control node - laptops, shared desktops, and servers can all run Ansible. You can have multiple control nodes, but Ansible itself does not coordinate across them, see AAP for a.

```

Also I want to ask what does the writer mean by "see AAP for a"

Reviewer: and the first sentence is a little bit hard to understand

Managed nodes
=============

> The target devices (servers, network appliances or any computer) you manage with Ansible. Managed nodes are also sometimes called "hosts". Ansible is not installed on managed nodes.

Reviewer: ```suggestion
With Ansible, you manage the target devices (servers, network appliances, or any computer). Managed nodes are also sometimes called "hosts". Ansible is not installed on managed nodes.
```

Inventory
=========

> A list of managed nodesi provided by one or more 'inventory sources'. Your inventory can specify information like IP address for each managed node,.organize managed nodes into groups and assigning data specific to the group or host by using variables. To learn more about inventory, see :ref:`the Working with Inventory<intro_inventory>` section. Sometimes an inventory source file is also reffered to as a 'hostfile'.

Reviewer: ```suggestion
A list of managed nodes provided by one or more 'inventory sources'. Your inventory can specify information like IP address for each managed node. Organize managed nodes into groups and assign data specific to the group or host using variables. See:ref:`the Working with Inventory<intro_inventory>` section to learn more about inventory. Sometimes an inventory source file is also referred to as a 'hostfile'.
```

Playbooks
=========

> They contain Plays (which are the basic unit of Ansible exeuction), this is both an 'execution concept' and how we describe the files on which ``ansible-playbook`` operates on.

Reviewer: ```suggestion
They contain Plays (the basic unit of Ansible execution), this is both an 'execution concept' and how we describe the files on which ``ansible-playbook`` operates.
		Playbooks are written in YAML and are easy to read, write, share and understand. To learn more about playbooks, see :ref:`about_playbooks`.
```

also where it goes "this is both an `execution concept`", what is both an execution concept ?

Playbooks are written in YAML and are easy to read, write, share and understand. To learn more about playbooks, see :ref:`about_playbooks`.

Plays
-----

> The playbook object that maps manged nodes (hosts) to tasks, all Ansible execution is run in this context. Contains an ordered list of tasks and can be run repeatedly and loop over the mapped hosts over each task.

Reviewer: ```suggestion
The playbook object maps managed nodes (hosts) to tasks. All Ansible execution runs in this context. Contains an ordered list of tasks and can be run repeatedly and looped over the mapped hosts over each task.
Plays can include variables as well as tasks.
```

"Contains an", what contains an ?

Plays can include variables as well as tasks.

Roles
.....

> A limited distribution of reusable ansible content (tasks, handlers, variables, plugins, templates and files) that can be referenced inside a Play.

Reviewer: ```suggestion
A limited distribution of reusable ansible content (tasks, handlers, variables, plugins, templates, and files) can be referenced inside a Play.
```

Tasks
.....

> The units of action in Ansible, must be contained in a Play, directly or via another playbook ojbect contained in a Play (Role, or imported/included task list file).

Reviewer: ```suggestion
The units of action in Ansible must be contained in a Play, directly or via another playbook object contained in a Play (Role, or imported/included task list file).
You can execute a single task once with an ad hoc command ``ansible`` or ``ansible-console`` (both create a virtual Play).
```

You can execute a single task once with an ad hoc command ``ansible`` or ``ansible-console`` (both create a virtual Play).

Handlers
........
Tasks that only execute when notified by a previous task which resulted in a 'changed' status.

Modules
=======

The units of code Ansible executes. Each module has a particular use, from administering users on a specific type of database to managing VLAN interfaces on a specific type of network device. You can invoke a single module with a task, or invoke several different modules in a playbook. Starting in Ansible 2.10, modules are grouped in collections. For an idea of how many collections Ansible includes, take a look at the :ref:`list_of_collections`.

Collections
===========

> Collections are a distribution format for Ansible content that can include playbooks, roles, modules, and plugins. You can install and use collections through `Ansible Galaxy <https://galaxy.ansible.com>`_. To learn more about collections, see :ref:`collections`. Unlike roles you can use any part of a collection without importing the whole thing.

Reviewer: ```suggestion
Collections are a distribution format for Ansible content that can include playbooks, roles, modules, and plugins. Unlike roles, you can use any part of a collection without importing the whole thing. You can install and use collections through `Ansible Galaxy <https://galaxy.ansible.com>`_. To learn more about collections, see :ref:`collections`.	
```

> AAP

Reviewer: sandra mentioned how we dont typically mention AAP in the docs, i just wanted to draw attention to this

===

> Ansilbe Automation Platform, this is a product that includes enteprise level features and integrates many tools of the Ansible ecosystem: ansible-core, awx, automation hub, etc.

Reviewer: ```suggestion
Ansible Automation Platform is a product that includes enterprise-level features and integrates many tools of the Ansible ecosystem: ansible-core, awx, automation hub, etc.
```

Control node
============
The machine from which you run the Ansible CLI tools (``ansible-playbook`` , ``ansible``, ``ansible-vault`` and others).
You can use any computer that meets the software requirements as a control node - laptops, shared desktops, and servers can all run Ansible.

> Multiple control nodes are possible, but Ansible itself does not coordinate across them, see ``AAP`` for such features.

Reviewer 2: ```suggestion
Multiple control nodes are possible, but Ansible itself does not coordinate across them.
```

Managed nodes
=============

> Also referred to as 'hosts', these are The target devices (servers, network appliances or any computer) you aim to manage with Ansible.

Reviewer 2: ```suggestion
Also referred to as 'hosts', these are the target devices (servers, network appliances or any computer) you aim to manage with Ansible.
```

Ansible is not normally installed on managed nodes, unless you are using ``ansible-pull``, but this is rare and not the recommended setup.

Inventory
=========
A list of managed nodes provided by one or more 'inventory sources'. Your inventory can specify information specific to each node, like IP address.
It is also used for assigning groups, that both allow for node selection in the Play and bulk variable assignment.
To learn more about inventory, see :ref:`the Working with Inventory<intro_inventory>` section. Sometimes an inventory source file is also referred to as a 'hostfile'.

Playbooks
=========

> They contain Plays (which are the basic unit of Ansible execution), this is both an 'execution concept' and how we describe the files on which ``ansible-playbook`` operates on.

Reviewer 2: ```suggestion
They contain Plays (which are the basic unit of Ansible execution). this is both an 'execution concept' and how we describe the files on which ``ansible-playbook`` operates on.
```

Playbooks are written in YAML and are easy to read, write, share and understand. To learn more about playbooks, see :ref:`about_playbooks`.

Plays
-----
The main context for Ansible execution, this playbook object maps managed nodes (hosts) to tasks.
The Play contains variables, roles and an ordered lists of tasks and can be run repeatedly.
It basically consists of an implicit loop over the mapped hosts and tasks and defines how to iterate over them.

Roles
.....
A limited distribution of reusable ansible content (tasks, handlers, variables, plugins, templates and files) for use inside of a Play.
To use any Role resource, the Role itself must be imported into the Play.

Tasks
.....
The definition of an 'action' to be applied to the managed host. Tasks must always be contained in a Play, directly or indirectly (Role, or imported/included task list file).
You can execute a single task once with an ad hoc command ``ansible`` or ``ansible-console`` (both create a virtual Play).

Handlers
........
Special form of a Task, that only execute when notified by a previous task which resulted in a 'changed' status.

Modules
=======

> The code or binaries that Ansible copies and executes on each managed node (when needed) to accoplish the action defined in each Task.

Reviewer 2: ```suggestion
The code or binaries that Ansible copies and executes on each managed node (when needed) to accomplish the action defined in each Task.
```

Each module has a particular use, from administering users on a specific type of database to managing VLAN interfaces on a specific type of network device.
You can invoke a single module with a task, or invoke several different modules in a playbook.

> Starting in Ansible 2.10, modules are grouped in collections. For an idea of how many collections Ansible includes, take a look at the :ref:`list_of_collections`.

Reviewer 2: ```suggestion
Ansible modules are grouped in collections. For an idea of how many collections Ansible includes, see the :ref:`list_of_collections`.
```

Plugins
=======
Pieces of code that expand Ansible's core capabilities, they can control how you connect to a managed node (connection plugins),
manipulate data (filter plugins) and even control what is displayed in the console (callback plugins).

> :ref:`working with plugins`

Reviewer 2: ```suggestion
:ref:`working_with_plugins`
```

> :ref:`working_with_plugins`

Reviewer 2: ```suggestion
See :ref:`working_with_plugins` for details.
```

Collections
===========

> Collections are a distribution format for Ansible content that can include playbooks, roles, modules, and plugins. You can install and use collections through `Ansible Galaxy <https://galaxy.ansible.com>`_. To learn more about collections, see :ref:`collections`. Colleciton resources can be used independantly and discretely from each other.

Reviewer 2: ```suggestion
Collections are a distribution format for Ansible content that can include playbooks, roles, modules, and plugins. You can install and use collections through `Ansible Galaxy <https://galaxy.ansible.com>`_. To learn more about collections, see :ref:`collections`. Collection resources can be used independently and discretely from each other.
```

AAP
===

> Short for 'Ansible Automation Platform'. This is a product that includes enterprise level features and integrates many tools of the Ansible ecosystem: ansible-core, awx, automation hub, etc.

Reviewer 2: ```suggestion
Short for 'Ansible Automation Platform'. This is a product that includes enterprise level features and integrates many tools of the Ansible ecosystem: ansible-core, awx, galaxyNG, and so on.
```
